Mathlib/Algebra/Order/LatticeGroup.lean

```diff
@@ -73,7 +81,8 @@ theorem mul_sup [CovariantClass α α (· * ·) (· ≤ ·)] (a b c : α) : c *
 #align add_sup add_sup
 
 @[to_additive]
-theorem sup_mul [CovariantClass α α (· * ·) (· ≤ ·)] (a b c : α) : (a ⊔ b) * c = a * c ⊔ b * c :=
+theorem sup_mul [CovariantClass α α (swap (· * ·)) (· ≤ ·)] (a b c : α) :
```

Comment:
Is there a version of this lemma that works both for `[Lattice α] [Group α] [CovariantClass _ _ _ _]` and `[LinearOrderedCommMonoid α]`?

Reply:
> This PR is an incremental step towards https://github.com/leanprover-community/mathlib4/issues/6376

My thinking was to do the refactor after merging this.

Reply:
Apologies - I've just realised you were asking about commutative monoids rather than groups. The proof here depends on a multiplicative inverse. I can add a note to the issue to consider further.

Now I think about it, I'm not sure if much of this requires associativity?